<a href="https://colab.research.google.com/github/TracyK10/flyrank-ml-internship-starter/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

I am choosing **Lane 2: Refresh / Content Opportunity Scoring**.

This lane produces a practical, prioritized review queue with explicit reason codes. That matters because human editors have limited time — they cannot manually dig through dashboards to find every page that deserves attention. A ranked queue tells them *which pages to look at first* and *why*, so they spend hours on high-impact updates instead of low-value noise.

Data and ML help here because many observable signals — impressions, clicks, position, sessions, engagement, and freshness — interact in ways too tangled for a single dashboard filter. The output is **decision-support**: it ranks candidates for review; it does not guarantee that editing a page will recover traffic.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

**Research question:** Which visible pages are showing evidence of true decline (not just seasonal noise) and should be prioritized for immediate content review?

| Framing element | Answer |
|---|---|
| **Decision** | Whether to allocate limited human editorial capacity to review and update a specific URL |
| **Who acts + action** | A human reviewer updates the page — for example, rewriting the title, expanding thin content, or fixing engagement issues |
| **Cost of a wrong call** | Wasting valuable editorial hours on a page experiencing normal seasonal shifts or consolidated traffic, while ignoring high-value pages that are silently bleeding real demand |
| **Task type** | Ranking / scoring — a priority-ordered review queue |
| **Target (starter proxy)** | Pages with an observable decline signal (`trend_direction == "down"`) — a current-window proxy, not a future outcome |
| **Success metric** | Precision@50 — of the top 50 flagged pages, how many are truly declining? |
| **Grain** | One row = one pseudonymized content item (page), aggregated over a trailing 90-day window |

**One-paragraph frame:**

> For human editors, deciding which page to review first, we will build a ranked review queue from observable search and engagement signals in the starter dataset, scoring decline risk measured by precision@50. A wrong call costs wasted editorial hours on seasonal noise while real declines are missed. A plain dashboard filter is not enough because many signals interact across visibility, freshness, position, and engagement. We will claim only observed, directional, decision-support results.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [2]:
import os
from pathlib import Path

import pandas as pd

# Resolve repo root when the notebook runs from work/notebooks/
if not Path("data/raw/content_refresh_anonymized.csv").exists():
    if Path("../../data/raw/content_refresh_anonymized.csv").exists():
        os.chdir("../..")

df = pd.read_csv("/content/content_refresh_anonymized.csv")

n_pages = len(df)
declining_share = (df["trend_direction"].str.lower() == "down").mean()
visible_pages = (df["impressions_90d"] >= 500).sum()

print(f"Pages in starter slice: {n_pages:,}")
print(f"Share with trend_direction == 'down': {declining_share:.1%}")
print(f"Pages with impressions_90d >= 500: {visible_pages:,}")

Pages in starter slice: 30,000
Share with trend_direction == 'down': 54.2%
Pages with impressions_90d >= 500: 16,726


These numbers show there is enough data volume to build a ranking queue:

- **30,000 pages** gives a workable inventory to score and rank, not just a toy sample.
- **54.2% flagged as declining** (`trend_direction == "down"`) means there are enough positive examples to train and evaluate a review queue — editors are not hunting needles in a haystack of zeros.
- **16,726 pages with ≥ 500 impressions** are visibly exposed enough that editorial time spent on them could matter.

The starter pipeline's verified results support this lane further: the transparent rule baseline achieves **Precision@50 = 0.240** (~12 of the top 50 correct), while the random forest reaches **Precision@50 = 0.740** (~37 of the top 50 correct). That gap suggests a learned ranking can beat a fixed rule on this slice — though these metrics come from the 30k starter dataset with client-holdout validation, not the full warehouse.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

**I can claim:**

- Which pages have the statistical footprint of a declining asset and deserve a reviewer's eyes first, based on historical observable patterns (impressions, clicks, position, sessions, engagement).

**I cannot claim:**

- That rewriting these pages will guarantee a traffic recovery — this is purely observational data, not a controlled causal experiment.
- That the starter `trend_direction` label proves future decline — it is a current-window proxy, not a forward-looking outcome.
- Anything about Google algorithm factors, AI citations, or product decision flags (those are not in the dataset).

## Self-check

**Capstone framing (per the lane guide):**

- [x] **Problem:** Prioritizing which pages deserve editorial review first amid limited capacity
- [x] **Decision:** Allocate reviewer time to a specific URL (yes/no / rank order)
- [x] **Grain:** One row = one pseudonymized content item (page-level, 90-day aggregates)
- [x] **Safe claims:** Decision-support and observed/directional language only; no causal recovery guarantees
- [x] **Observable signals only:** No product decision flags used as features or truth

**Submission checklist:**

- [x] Every section above is filled — markdown thinking AND the code that backs it
- [x] The notebook runs top to bottom with no errors (Runtime → Run all)
- [x] No client names, URLs, or private queries anywhere
- [x] My claims use careful words: observed, measured, directional, decision-support
- [x] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.